File: Back/.Net/BetFriends.Api/Features/Bets/CompleteBetController.cs
using BetFriends.Application.Abstractions;
using BetFriends.Application.Features.CompleteBet;
using BetFriends.Domain.Bets.Exceptions;
using Microsoft.AspNetCore.Mvc;

namespace BetFriends.Api.Features.Bets;

public class CompleteBetController : Controller
{
    private readonly IBetModule betModule;
    private readonly CompleteBetPresenter completeBetPresenter;
    private readonly ILogger<CompleteBetController> logger;

    public CompleteBetController(IBetModule betModule,
                                 CompleteBetPresenter completeBetPresenter,
                                 ILogger<CompleteBetController> logger)
    {
        this.betModule = betModule;
        this.completeBetPresenter = completeBetPresenter;
        this.logger = logger;
    }

    [HttpPost("bets/{betId:guid}/complete")]
    public async Task<IActionResult> Complete([FromBody] CompleteBetInput completeBetInput, Guid betId)
    {
        try
        {
            await betModule.ExecuteAsync(new CompleteBetCommand(betId, completeBetInput.IsSuccessful, completeBetInput.Proof));
        }
        catch (BookieCompleteException bex)
        {
            completeBetPresenter.BookieError(bex.Message);
        }
        catch(ProofRequiredException prex)
        {
            completeBetPresenter.ProofRequired(prex.Message);
        }
        return completeBetPresenter.ViewModel;
    }
}

public class CompleteBetPresenter : ICompleteBetOutputPort
{
    public IActionResult ViewModel { get; private set; } = null!;

    public void BetDoesNotExist(Guid betId)
    {
        ViewModel = new NotFoundObjectResult(betId);
    }

    public void Success(CompleteBetResponse completeBetResponse)
    {
        ViewModel = new OkResult();
    }

    internal void BookieError(string message)
    {
        ViewModel = new BadRequestObjectResult(message);
    }

    internal void ProofRequired(string message)
    {
        ViewModel = new BadRequestObjectResult(message);
    }
}

public record CompleteBetInput(bool IsSuccessful, string? Proof);

File: Back/NodeJS/src/infrastructure/api/nest/src/bet/features/complete-bet/CompleteBet.controller.ts
import { Body, Controller, Inject, Param, Post, Res } from "@nestjs/common";
import { IBetModule } from "../../../../../../../application/Abstractions/IBetModule";
import { CompleteBetInput } from "./CompleteBetInput";
import { FastifyReply } from 'fastify';
import { CompleteBetPresenter } from "./CompleteBetPresenter";
import { CompleteBetCommand } from "../../../../../../../application/features/complete-bet/CompleteBetHandler";

@Controller('bets')
export class CompleteBetController {
    constructor(@Inject('IBetModule') private betModule: IBetModule,
                private presenter: CompleteBetPresenter) {}

    @Post(':betId/complete')
    async Complete(@Body() input: CompleteBetInput, @Param('betId') betId: string, @Res() res: FastifyReply) {
        const command = new CompleteBetCommand(betId, input.isSuccessful, input.proof);
        await this.betModule.ExecuteCommand(command);
        return this.presenter.BuildResponse(res);
    }
}